includes/helpers.php

```diff
@@ -105,3 +105,25 @@ function array_pairs($array)
         return $pairs;
     }
 }
+
+/**
+ * Cast string to int or float.
+ * Returns 0 if string is not numeric
+ *
+ * @param string $number
+ * @return float|int
+ */
+function cast_number($number)
+{
+    if (! is_numeric($number)) {
+        if (! preg_match('/-?\d+(\.\d+)?/', $number, $matches)) {
```

Comment:
This could have some weird side-effects since it would extract the first number from any position in a string.  
Maybe that should be its own function since its a bit more than "casting"

Reply:
@Jellyfrog actually, the only reason it is there is to match legacy behavior...  `'23sdfadf' + 0 === 23`

Reply:
Horrifying :)